Makefile

```diff
@@ -40,10 +40,26 @@ from-scratch: ## start development environment from scratch
 	make up
 
 devstack-install: ## install local version of package into docker devstack for development
-	docker exec -t edx.devstack.lms bash -c 'source /edx/app/edxapp/nodeenvs/edxapp/bin/activate && cd /edx/app/edxapp/edx-platform && npm uninstall @edx/studio-frontend && cd /edx/src/studio-frontend && npm link && cd /edx/app/edxapp/edx-platform && npm link @edx/studio-frontend'
+	# note that this assumes a structure /src/sfe_parent/studio-frontend
+	# step 0: remove studio-frontend from edx-platform/node_modules
+	docker exec -t edx.devstack.studio bash -c 'source /edx/app/edxapp/nodeenvs/edxapp/bin/activate && cd /edx/app/edxapp/edx-platform && npm uninstall @edx/studio-frontend && rm -rf ./node_modules/@edx/studio-frontend'
+	# step 1: create symlink in /edx/src/sfe_parent/symlinks
+	# note that we delete studio-frontend/node_modules; this is deliberate
+	docker exec -t edx.devstack.studio bash -c 'mkdir -p /edx/src/sfe_parent/symlinks && rm -rf /edx/src/sfe_parent/symlinks/* && source /edx/app/edxapp/edxapp_env && cd /edx/src/sfe_parent/studio-frontend && export npm_config_prefix=/edx/src/sfe_parent/symlinks && npm link && rm -rf /edx/src/sfe_parent/studio-frontend/node_modules'
```

Comment:
Does npm link use the npm_config_prefix?

Reply:
Yes - [link 1](https://docs.npmjs.com/cli/link#description), [link 2](https://docs.npmjs.com/misc/config#prefix)

```
First, npm link in a package folder will create a symlink in the global folder {prefix}/lib/node_modules/<package> that links to the package where the npm link command was executed. (see npm-config for the value of prefix). It will also link any bins in the package to {prefix}/bin/{name}.
```

Reply:
Thanks!

I could very well be misunderstanding something, but where does the studio-frontend subdirectory of sfe_parent get created? I see that you note that it assumes the structure /src/sfe_parent/studio-frontend, but where is that created?  Is that something we need to enforce in order for this to work? I couldn't find any other reference to it elsewhere, so I assume you'd have to manually create it before running this command? Could you clarify?

Reply:
> so I assume you'd have to manually create it before running this command

This is how I "solved" the chicken/egg problem of needing to describe the folder structure that contains studio-frontend in a file inside studio-frontend. I usually create directories with git clone, like this:

```
(after starting devstack containers)
ericfischer@Eric-Fischers-MacBook-Pro:~/work/devstack$ cd ../src/
ericfischer@Eric-Fischers-MacBook-Pro:~/work/src$ mkdir sfe_parent
ericfischer@Eric-Fischers-MacBook-Pro:~/work/src$ cd sfe_parent
ericfischer@Eric-Fischers-MacBook-Pro:~/work/src/sfe_parent$ git clone git@github.com:edx/studio-frontend.git
...
ericfischer@Eric-Fischers-MacBook-Pro:~/work/src/sfe_parent$ cd studio-frontend
ericfischer@Eric-Fischers-MacBook-Pro:~/work/src/sfe_parent/studio-frontend$ git checkout efischer/dependency_funtimes
...
ericfischer@Eric-Fischers-MacBook-Pro:~/work/src/sfe_parent/studio-frontend$ make devstack-install
```

Is there any easy way (in Makefile) to check the structure and error out if it's not correct? I'll look into that further

Reply:
Ah, I see. Could this be useful (second answer) https://stackoverflow.com/questions/5553352/how-do-i-check-if-file-exists-in-makefile? The wildcard could be promising.

Reply:
Were you able to look into whether it would be worthwhile to verify the directory structure is correct to avoid issues? Or should we add a make target somewhere to do this work for the user?

Reply:
Ah yes, I had forgotten about that :/

I'll get something up to address it.